Note:
Carry over the patch for udev 175 detailed below for eudev 3.0+ but applied to rules/60-block.rules - Anto

From: Martin Pitt <martin.pitt@ubuntu.com>
Date: Sat, 26 Apr 2014 16:34:06 +0200
Subject: Set default polling interval on removable devices as well

The events_dfl_poll_msecs rule will not trigger if "block" is not a module, but
built in. This will avoid udisks etc. having to poll from userspace, and
provide proper ejection when the hardware eject button is pressed.

Bug-Debian: https://bugs.debian.org/713877
Bug-Ubuntu: https://launchpad.net/bugs/890592
---
 rules/60-block.rules | 1 +
 1 file changed, 1 insertion(+)

diff --git a/rules/60-block.rules b/rules/60-block.rules
index de41499..1b5e294 100644
--- a/rules/60-block.rules
+++ b/rules/60-block.rules
@@ -3,6 +3,7 @@
 # enable in-kernel media-presence polling
 ACTION=="add", SUBSYSTEM=="module", KERNEL=="block", ATTR{parameters/events_dfl_poll_msecs}=="0", \
   ATTR{parameters/events_dfl_poll_msecs}="2000"
+ACTION=="add", ATTR{removable}=="1", ATTR{events_poll_msecs}=="-1", ATTR{events_poll_msecs}="2000"
 
 # forward scsi device event to corresponding block device
 ACTION=="change", SUBSYSTEM=="scsi", ENV{DEVTYPE}=="scsi_device", TEST=="block", ATTR{block/*/uevent}="change"

